### PR TITLE
feat: sparql string interpolator to resolve iterables considering context

### LINKED ISSUE #1399: Sparql query string interpolator to use proper collection encoding
As a renku graph developer, I'd like the SPARQL query string interpolator to be smart enough to find out what sort of collection encoding needs to be used in a particular context of the part of the query.

Acceptance criteria:

- [x] the interpolator to encode collection items as values wrapped in parenthesis and separated with a '` `' (space) if in the context of the `VALUES` clause
- [x] the interpolator to encode collection items as values separated with a '`,`' (comma) if in the context of the `IN` clause

### PR DESCRIPTION
**Pull request by jachro**
closes #1399

### CODE REVIEW

File: triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    sealed trait ClauseType
+    object ClauseType {
+      case object VALUES extends ClauseType
+      case object IN     extends ClauseType
+    }
+
+    def detectClauseContext(snippet: String): Option[ClauseType] = {
+      val validated = snippet.filter(c => c >= ' ' && c != '|').toUpperCase
+      if (valuesClause matches validated) Some(ClauseType.VALUES)
```
Comment (eikek) [inline triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala]: Wondering what is when there is an `IN` and a `VALUES` in the query where the `IN` comes last. Wouldn't it return in `ClauseType.VALUES`? I thought perhaps it is enough to simply look backwards (maybe via `String#lastIndexOf` and selecting the larger index).

**Review by eikek — APPROVED** [review]

File: triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala (hunk 1 above)
Comment (jachro) [reply triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala]: I think we should be fine as the regex is matching at the end of the line. I can maybe write a test verifying that.

Comment (eikek) [reply triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala]: Yes, but isn't there a word matcher `\\w+` that could interfere here? not sure, regex is always a bit hard to verify by eye for me 😄 

Another note, again not sure: couldn't we make the regex simpler by using one class with all possible characters or is the exact sequence important? like `[\\s\\(\\?\\)]*` The single character classes `[\\s]` look unnecessary - why not only `\\s`?

Comment (eikek) [reply triples-store-client/src/main/scala/io/renku/triplesstore/client/sparql/StringInterpolator.scala]: Ah I think I see now, it wants to match the exact phrase until the open curly. Forget my comment above.

**Pull request merged**